core/src/com/unciv/logic/map/UnitMovementAlgorithms.kt

```diff
@@ -121,52 +124,51 @@ class UnitMovementAlgorithms(val unit:MapUnit) {
                     path.add(intermediateTile)
                     intermediateTile = movementTreeParents[intermediateTile]!!
                 }
-                return path.reversed() // and reverse in order to get the list in chronological order
+                path.reverse() // and reverse in order to get the list in chronological order
+                return path
             }
 
             if (newTilesToCheck.isEmpty()) return emptyList() // there is NO PATH (eg blocked by enemy units)
 
             // no need to check tiles that are surrounded by reachable tiles, only need to check the edgemost tiles.
             // Because anything we can reach from intermediate tiles, can be more easily reached by the edgemost tiles,
-            // since we'll have to pass through an edgemost tile in order to reach the diestination anyway
-            tilesToCheck = newTilesToCheck.filterNot { tile -> tile.neighbors.all { newTilesToCheck.contains(it) || tilesToCheck.contains(it) } }
+            // since we'll have to pass through an edgemost tile in order to reach the destination anyway
+            tilesToCheck = newTilesToCheck.filterNot { tile -> tile.neighbors.all { it in newTilesToCheck || it in tilesToCheck } }
 
             distance++
         }
     }
 
-    fun getTileToMoveToThisTurn(finalDestination:TileInfo): TileInfo {
+    fun getTileToMoveToThisTurn(finalDestination: TileInfo): TileInfo {
 
         val currentTile = unit.getTile()
         if (currentTile == finalDestination) return currentTile
 
-        if(unit.type.isAirUnit()){
-            return finalDestination // head there directly
-        }
-
+        // head there directly
+        if (unit.type.isAirUnit()) return finalDestination
 
         val distanceToTiles = getDistanceToTiles()
 
-        if (distanceToTiles.containsKey(finalDestination)) { // we should be able to get there this turn
-            if (canMoveTo(finalDestination))
-                return finalDestination
+        class UnreachableDestinationException : Exception()
 
-            // Someone is blocking to the path to the final tile...
-            val destinationNeighbors = finalDestination.neighbors
-            if (destinationNeighbors.contains(currentTile)) // We're right nearby anyway, no need to move
-                return currentTile
+        // If the tile is far away, we need to build a path how to get there, and then take the first step
+        if (!distanceToTiles.containsKey(finalDestination))
+            return getShortestPath(finalDestination).firstOrNull()
+                    ?: throw UnreachableDestinationException()
```

Comment:
inverted the if for less nesting. getShortestPath(finalDestination).firstOrNull() is the same as the stuff below where you get shortest path, then check for empty, the return first